db/mongo.go

```diff
@@ -1,43 +1,46 @@
-package mongo
+package db
 
 import (
+	"sync"
+
 	"github.com/flavioribeiro/gonfig"
 	"github.com/snickers/snickers/types"
 	"gopkg.in/mgo.v2"
 	"gopkg.in/mgo.v2/bson"
 )
 
 // Database struct that persists configurations
-type Database struct {
+type mongoDatabase struct {
 	db *mgo.Database
 }
 
-var instance *Database
+var databaseMongoInit sync.Once
+var mongoInstance *mongoDatabase
 
 // GetDatabase returns database singleton
-func GetDatabase(configPath string) (*Database, error) {
-	instance = &Database{}
-	cfg, err := gonfig.FromJsonFile(configPath)
-	if err != nil {
-		return nil, err
-	}
-	mongoHost, _ := cfg.GetString("MONGODB_HOST", "")
-	session, err := mgo.Dial(mongoHost)
-	if err != nil {
-		return &Database{}, err
-	}
-	session.SetMode(mgo.Monotonic, true)
-	instance.db = session.DB("snickers")
-	return instance, nil
+func getMongoDatabase(cfg gonfig.Gonfig) (Storage, error) {
+	var initErr error
+	databaseMongoInit.Do(func() {
+		mongoInstance = &mongoDatabase{}
+		mongoHost, _ := cfg.GetString("MONGODB_HOST", "")
+		session, err := mgo.Dial(mongoHost)
+		if err != nil {
+			initErr = err
```

Comment:
can you write a test for the case when mongo fails? like trying to connect to a random IP.. just to cover this `if`